components/org.wso2.carbon.healthcheck.api.core/src/main/java/org/wso2/carbon/healthcheck/api/core/impl/SuperTenantUSHealthChecker.java

```diff
@@ -136,8 +139,28 @@ protected void validateUSM(UserStoreManager userStoreManager, String domainName,
             if (log.isDebugEnabled()) {
                 log.debug("Error while checking existence of user in user store " + domainName, e);
             }
-            errors.add(new HealthCheckError(Constants.ErrorCodes.ERROR_USERSTORE_CONNECTIVITY_BY_IS_USER_EXISTING
-                    , "Error while checking health of USM with domain: " + domainName,
+            errors.add(new HealthCheckError(Constants.ErrorCodes.ERROR_USERSTORE_CONNECTIVITY_BY_IS_USER_EXISTING,
+                    "Error while checking health of USM with domain: " + domainName,
+                    Utils.getRootCauseMessage(e)));
+        }
+    }
+
+    protected void validateUSM(UserStoreManager userStoreManager, String domainName, Properties properties,
```

Comment:
AFAIK the implementation of previous method is wrong right? If so, shall we deprecate the previous method? And add a method comment to the new method? WDYT?

Reply:
+1. I'll do that.